### DISCUSSION TITLE
Get triggers/info from CMake Tools extension in my own extension

### DISCUSSION THREAD

**Discussion by itavero** [microsoft/vscode-cmake-tools#3114] [Q&A] [answered]
I'm thinking about writing an extension which will need to interact with the CMake cache (for a CMake based dependency manager we are using).
Can I somehow get a trigger from this extension when the CMake configuration phase is done (e.g. the  CMake cache might be updated)?

Additionally I would also like to be able to figure out where the CMakeCache.txt file is (e.g. what the build directory is that is being used by this extension).

I'm new to VS Code Extension development, so I'm not sure what the possibilities are for retrieving such information/triggers. Hopefully someone can point me in the right direction!

**Comment by itavero** [comment]
In the meantime I became aware of the vscode-cmake-tools-api package. It seems that it should be possible to figure out what the build directory is and thus where I can find the `CMakeCache.txt`.
What I find a bit unclear is how I can discover what the current active project is, as I only see events for when it changes.

**Comment by gcampbell-msft** [comment, marked as answer]
@itavero The most likely answer would be to interact with the vscode-cmake-tools-api package. To get the active project, one way would be to get the active folder using the `getActiveFolderPath` method and then pass that into `getProject`.
